cli/src/nonce.rs

```diff
@@ -0,0 +1,466 @@
+                ),
+        )
+    }
+}
+
+pub fn parse_nonce_create_account(matches: &ArgMatches<'_>) -> Result<CliCommandInfo, CliError> {
+    let nonce_account = keypair_of(matches, "nonce_account_keypair").unwrap();
+    let lamports = amount_of(matches, "amount", "unit").expect("Invalid amount");
+
+    Ok(CliCommandInfo {
+        command: CliCommand::CreateNonceAccount {
+            nonce_account: nonce_account.into(),
+            lamports,
+        },
+        require_keypair: true,
+    })
+}
+
+pub fn parse_new_nonce(matches: &ArgMatches<'_>) -> Result<CliCommandInfo, CliError> {
+    let nonce_account = keypair_of(matches, "nonce_account_keypair").unwrap();
+
+    Ok(CliCommandInfo {
+        command: CliCommand::NewNonce(nonce_account.into()),
+        require_keypair: true,
+    })
+}
+
+pub fn parse_get_nonce(matches: &ArgMatches<'_>) -> Result<CliCommandInfo, CliError> {
+    let nonce_account_pubkey = pubkey_of(matches, "nonce_account_pubkey").unwrap();
+
+    Ok(CliCommandInfo {
+        command: CliCommand::GetNonce(nonce_account_pubkey),
+        require_keypair: false,
+    })
+}
+
+pub fn parse_withdraw_from_nonce_account(
+    matches: &ArgMatches<'_>,
+) -> Result<CliCommandInfo, CliError> {
+    let nonce_account = keypair_of(matches, "nonce_account_keypair").unwrap();
+    let destination_account_pubkey = pubkey_of(matches, "destination_account_pubkey").unwrap();
+    let lamports = amount_of(matches, "amount", "unit").expect("Invalid amount");
+
+    Ok(CliCommandInfo {
+        command: CliCommand::WithdrawFromNonceAccount {
+            nonce_account: nonce_account.into(),
+            destination_account_pubkey,
+            lamports,
+        },
+        require_keypair: true,
+    })
+}
+
+pub fn parse_show_nonce_account(matches: &ArgMatches<'_>) -> Result<CliCommandInfo, CliError> {
+    let nonce_account_pubkey = pubkey_of(matches, "nonce_account_pubkey").unwrap();
+    let use_lamports_unit = matches.is_present("lamports");
+
+    Ok(CliCommandInfo {
+        command: CliCommand::ShowNonceAccount {
+            nonce_account_pubkey,
+            use_lamports_unit,
+        },
+        require_keypair: false,
+    })
+}
+
+pub fn process_create_nonce_account(
+    rpc_client: &RpcClient,
+    config: &CliConfig,
+    nonce_account: &Keypair,
+    lamports: u64,
+) -> ProcessResult {
+    let nonce_account_pubkey = nonce_account.pubkey();
+    check_unique_pubkeys(
+        (&config.keypair.pubkey(), "cli keypair".to_string()),
+        (&nonce_account_pubkey, "nonce_account_pubkey".to_string()),
+    )?;
+
+    if rpc_client.get_account(&nonce_account_pubkey).is_ok() {
+        return Err(CliError::BadParameter(format!(
+            "Unable to create nonce account. Nonce account already exists: {}",
+            nonce_account_pubkey
+        ))
+        .into());
+    }
+
+    let minimum_balance =
+        rpc_client.get_minimum_balance_for_rent_exemption(std::mem::size_of::<NonceState>())?;
```

Comment:
Will `std::mem::size_of()` match `serialized_size()` here?  I had a conflict between the two at some point, hence `NonceState:size()`, which could be used here.